src/lib/homestore/heap_chunk_selector.cpp

```diff
@@ -0,0 +1,116 @@
+#include "heap_chunk_selector.h"
+
+#include <execution>
+#include <algorithm>
+#include <sisl/logging/logging.h>
+
+namespace homeobject {
+
+//https://github.com/facebook/folly/blob/61c11d77eb9a8bdc60f673017fccfbe900125cb6/folly/AtomicHashMap.h#L42
+//AtomichashMap has a max size limit of ~18x initial size.
+
+HeapChunkSelector::HeapChunkSelector() : 
+m_pdev_heap_map(PdevHeapMap(pdev_atomicmap_init_size)), 
+m_pdev_avalable_blk_map(pdev_atomicmap_init_size), 
+m_chunks(chunk_atomicmap_init_size) {}
+
+HeapChunkSelector::HeapChunkSelector(const uint32_t& pdev_heap_map_initial_size, 
+const uint32_t& pdev_avalable_blk_map_initial_size, const uint32_t& chunk_initial_num) : 
+m_pdev_heap_map(PdevHeapMap(pdev_heap_map_initial_size)), 
+m_pdev_avalable_blk_map(PdevAvalableBlkMap(pdev_avalable_blk_map_initial_size)),
+m_chunks(ChunkMap(chunk_initial_num)) {}
+
+
+//add_chunk may be called in Homeobject(when sealing a shard) or in Homestore(when adding chunks to vdev) concurently
+void HeapChunkSelector::add_chunk(csharedChunk& chunk) {
```

Comment:
IIRC we need a add::chunk(uint64 chunkID) overload for HO?

Reply:
we already has a `release_chunk` for this purpose.
https://github.com/eBay/HomeObject/blob/27e9f1de6ae598da51b55f71ff599575db5ef893/src/lib/homestore/heap_chunk_selector.cpp#L108

Reply:
sorry my bad. Yes
LGTM \